### DISCUSSION TITLE
Is it possible to change the input parameters step by step using step function?

### DISCUSSION THREAD

**Discussion by RuiheLi** [pybamm-team/PyBaMM#1545] [Q&A]
Hi~ I notice in this notebook (https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/solution-data-and-processed-variables.ipynb), we can 'step the solution and look at the results as we go'. So can we also change the input parameters based on the calculated results? In this way, we can renew some parameter values without making them a variable, and maybe save computational resource, as we can update these parameters every 100 cycles, rather than every seconds.

**Comment by valentinsulzer** [comment]
Yes, if you want. Which parameters are you thinking of? Let me know what you want to do and I can send an example

**Reply by RuiheLi** [reply]
Woo~ Wonderful! I think the EC concentration is my current target now.

**Comment by RuiheLi** [comment]
Just to let other people know, the 'step in' function, or called 'initialize-model-with-solution', can be found in https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/initialize-model-with-solution.ipynb
